### PR TITLE
File uploading using multer

### PR DESCRIPTION
**Pull request by panchalmilan**

### CODE REVIEW

**Review by namandangi — COMMENTED** [review]
@panchalmilan
1. Please explain your logic behind meta-data in brief.
2. supervisor, nodemon etc. are dev dependencies.
3. Did you implement logic for folder uploading using repoDirPath ?

**Comment by namandangi** [comment]
Always leave a brief description about the changes/feature that you've implemented in the PR.

**Comment by panchalmilan** [comment]
### Functionality in file uploads:
 when a user uploads any file for the first 
 * user's folder is created in uploads folder naming <username>
* metadata file is created in respective user's folder ie **base/uploads/<username>/metadata.txt**
 
     metadata contains all info regarding respective user's file crud
		-till now, it just says that if user has a) created new file b) updated a existed file,
                 
    Gives info abt which file added/change d with date and time
		   [this is done in base/utility/fileExistence.js]
	         <repo_name> folder is created for respective user ie **base/uploads/<username>/<repo_name>**

*All folder creation is done in base/middleware/dirCreation.js*

[wrote a lot of code for dir creation later realized it could be done easily using recursive creation :(  fs.mkdirSync(dir, {recursive: true})]

File: controllers/repository.js [hunk 1]
```diff
@@ -49,8 +59,21 @@ exports.updateRepo = async (req, res) => {
...
   }
 };
+
+exports.uploadFiles = async (req, res) => {
+  try {
+    if (!req.file) return res.send('No file uploaded');
+
+    // check if file already exists
+    fileExistence(req);
+
```
Comment (panchalmilan) [inline controllers/repository.js:74]: metadata.txt is updated with current file upload

File: middleware/dirCreation.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const path = require('path');
+const fs = require('fs');
+
+module.exports = function dirCreation(req, res, next) {
+  const { username, repo } = req.params;
+
+  console.log(username, repo);
+
```
Comment (panchalmilan) [inline middleware/dirCreation.js:8]: this function creates below dir structure by checking its existence

 /uploads
    /<username>
        metadata.txt
        /<repo_name>
            <file1.js>
            <file2.html>
        /<repo_name>
            <xfile1.js>
            <xfile2.html>
    /<username>
        ...

Comment (panchalmilan) [inline middleware/dirCreation.js:8]: this function creates folder in below structure by checcking each folders existence
[can be imporoved by using recursive creation]
```
 /uploads
    /<username>
        metadata.txt
        /<repo_name>
            <file1.js>
            <file2.html>
        /<repo_name>
            <xfile1.js>
            <xfile2.html>
    /<username>
        ...
```

File: utility/fileExistence.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const fs = require('fs');
+const path = require('path');
+
+exports.fileExistence = ({ file, params }) => {
+  const { username, repo } = params;
```
Comment (panchalmilan) [inline utility/fileExistence.js:5]: this functions appends metadata of file upload and modification by the user in respective repository
metadata includes
    1. reponame
    2. filename
    3. date and time of modification

**Pull request merged**